tests/state_machine.test.cpp

```diff
@@ -0,0 +1,112 @@
+#include <memory>
+#include <optional>
+#include <sstream>
+#include <type_traits>
+#include <typeindex>
+#include <vector>
+
+#include <boost/ut.hpp>
+
+#include "concrete_states.hpp"
+#include "state.hpp"
+#include "state_context.hpp"
+#include "state_machine.hpp"
+
+// Mock State for testing
+class MockState : public State
+{
+  public:
+    bool enter_called = false;
+    bool handle_called = false;
+    bool exit_called = false;
+
+    void enter(StateContext& context) override
+    {
+        std::ignore = context;
+        enter_called = true;
+    }
+
+    void handle(StateContext& context) override
+    {
+        std::ignore = context;
+        handle_called = true;
+    }
+
+    std::optional<std::unique_ptr<State>> exit(StateContext& context) override
+    {
+        std::ignore = context;
+        exit_called = true;
+        return std::nullopt;
+    }
+};
+
+void state_machine_tests()
+{
+    using namespace boost::ut;
+
+    "StateMachine"_test = [] {
+        "Initalize inital state"_test = [] {
+            StateMachine sm;
+            auto actual_state = sm.get_current_state();
+            expect(actual_state.has_value()) << "A State is initialized";
+
+            State& actual_state_ref = actual_state.value().get();
+            expect(typeid(actual_state_ref) == typeid(UserInputState))
```

Comment:
Please add `that %` in front of your boolean expressions. This is so, if it errors it shows you the actual expression that was not true. Without that it will just show `false` 

Example: https://godbolt.org/z/KMPr7ojfY